gitpod-io/gitpod
# Don't mark prebuilds with failing tasks unavailable

This PR correctly marks prebuilds with a failed headless task as available, and updates ws-manager to not spuriously add the `failed` condition if the headless task fails.

fixes #4964

### How to test
1. Run a prebuild with a failing headless task, e.g. https://csweichel-don-t-abort-prebuild-4964.staging.gitpod-dev.com#https://github.com/csweichel/test-repo/tree/task-failure
2. Check the database and find the prebuild marked as available

![image](https://user-images.githubusercontent.com/3210701/127274865-1bf67489-dabc-4032-9ef0-db87b6186c62.png)
